spring-mvc-basics-2/src/main/java/com/baeldung/spring/configuration/EmailConfiguration.java

```diff
@@ -53,19 +88,28 @@ public SpringTemplateEngine thymeleafTemplateEngine() {
     }
 
     @Bean
-    public SpringResourceTemplateResolver thymeleafTemplateResolver() {
-        SpringResourceTemplateResolver templateResolver = new SpringResourceTemplateResolver();
-        templateResolver.setPrefix("/WEB-INF/views/mail/");
+    public ITemplateResolver thymeleafTemplateResolver() {
+    	AbstractConfigurableTemplateResolver templateResolver;
+    	templateResolver = mailTemplatesExternal ? new FileTemplateResolver() : new ClassLoaderTemplateResolver();
+        templateResolver.setPrefix(mailTemplatesPath + "/");
         templateResolver.setSuffix(".html");
         templateResolver.setTemplateMode("HTML");
         templateResolver.setCharacterEncoding("UTF-8");
         return templateResolver;
     }
     
     @Bean 
-    public FreeMarkerConfigurer freemarkerConfig() { 
-        FreeMarkerConfigurer freeMarkerConfigurer = new FreeMarkerConfigurer(); 
-        freeMarkerConfigurer.setTemplateLoaderPath("/WEB-INF/views/mail");
+    public FreeMarkerConfigurer freemarkerConfig() throws IOException {
+        Configuration configuration = new Configuration(Configuration.VERSION_2_3_27);
+        if (mailTemplatesExternal) {
```

Comment:
This seems to go against the discussion where we said we wouldn't make this a configuration option so much as two implementations the reader could choose between.